### ISSUE TITLE
[feature request] More masked reductions: amin/amax, argmin/argmax

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#69359]
An important feature would be passing a custom "fill value" (for the case when no elements in the mask across a dimension are True), and that a fill value could be an appropriately sized (broadcastable) tensor. 

This  is also related to reducing over empty dimensions. If custom "fill values" are supported, all reductions across empty dimensions will make sense even for the cases when the output is not  an empty tensor

**Comment by vadimkantorov** [comment]
Currently I use the following, but I'm not sure if `torch.where(mask, x, 0)` is as fast as `torch.mul(x, mask)`. Maybe `torch.where` could dispatch to `torch.mul` if needed to gain more speed. `torch.where` could still be more expressive in the code for consistency

```python
masked_amax = lambda x, dim, mask, fill_value = 0  : torch.where(mask, x, x.new([fill_value])).amax(dim = dim)
masked_amin = lambda x, dim, mask, fill_value = 1e8: torch.where(mask, x, x.new([fill_value])).amin(dim = dim)
```

**Comment by vadimkantorov** [comment]
for `quantile` it seems quite impossible to have a hand-rolled vectorized masked impl

**Comment by vadimkantorov** [comment]
masked mean/std/friends are useful for creating a simple robust mean/std estimation that only takes mean between certain quantiles (and selecting only elements within the quantiles produces a mask)

**Comment by vadimkantorov** [comment]
Maybe `nan*` versions could be overturned for the purpose of general masked operators!

**Comment by cpuhrsch** [comment]
cc @george-qi for https://pytorch.org/maskedtensor/main/index.html

**Comment by pearu** [comment]
I believe https://github.com/pytorch/pytorch/pull/72723 tackles this issue.
